README.md

```diff
@@ -292,6 +292,7 @@ The available groups are:
 - `lint`: Required for linting and formatting.
 - `mypy`: Required for static type checking.
 - `onnx`: Required for using custom surrogate models in [ONNX format](https://onnx.ai).
+- `polars`: Required for optimized search space construction via [Polars](https://docs.pola.rs/)

```

Comment:
Hi @Scienfitz, general comment upfront: Unless part of type checking (i.e. `if TYPE_CHECKING`) I think you should import polars *through* a yet to be added `polars.py` module in `_optional`. This has the advantages that
* It's clear in the code that stuff is coming from an optional module
* You can centrally maintain the import error message there (see similar modules)
* And don't need to rely on the fact (which you currently do) that parts of the code where the import happens are shielded by some outer switching logic to pandas as fallback. That is, currently people would get a cryptic `ModuleNotFoundError` if by any chance there is an execution path that tries to execute your lazy imports, while in the alternative way they'd at least get the correct error message written by us.

Reply:
ive pushed the change as it makes sense, but still wonder why you would suggest what youve declined elsewhere

Reply:
for some reason my comment was posted before your last one even showed up for me

The main argument back thenw as that now on would have ti implement torch everywhere from a util module. But here I now have to do the same for polars. That seems inconsistent to me. But Ill resolve since the best solution for this PR is agreed